go/worker/storage/committee/checkpoint_sync.go

```diff
@@ -296,9 +331,36 @@ func (n *Node) getCheckpointList() ([]*storageSync.Checkpoint, error) {
 	return list, nil
 }
 
+// fetchCheckpoints fetches checkpoints using checkpoint sync p2p protocol client.
+//
+// In case of no peers, error or no checkpoints, it fallbacks to the legacy storage sync protocol.
+func (n *Node) fetchCheckpoints(ctx context.Context) ([]*checkpointsync.Checkpoint, error) {
+	list1, err := n.checkpointSync.GetCheckpoints(ctx, &checkpointsync.GetCheckpointsRequest{
+		Version: 1,
+	})
+	if err == nil && len(list1) > 0 { // if NO error and at least one checkpoint
+		return list1, nil
+	}
+
+	list2, err := n.legacyStorageSync.GetCheckpoints(ctx, &synclegacy.GetCheckpointsRequest{
+		Version: 1,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var cps []*checkpointsync.Checkpoint
```

Comment:
```suggestion
	cps := make([]*checkpointsync.Checkpoint, 0, len(list2))
```
I usually always create a slice with the right capacity, if the number of elements is known.

Reply:
I try to avoid indexes unless many elements (performance), as I find the code harder to read with indexes, so would prefer to keep as is but also happy to change!